includes/HttpFunctions.php

```diff
@@ -382,9 +382,13 @@ public function proxySetup() {
 			return;
 		}
 
-		global $wgHTTPProxy;
+		global $wgHTTPProxy, $wgSetOriginalHostHeader;
 		if ( $wgHTTPProxy ) {
-			$this->proxy = $wgHTTPProxy ;
+			$this->proxy = $wgHTTPProxy;
+			if ( $wgSetOriginalHostHeader ) {
```

Comment:
`!empty($wgSetOriginalHostHeader)` - this variable is not always set via `LocalSettings.php`.